### ISSUE TITLE
Result framework syntax change?

### ISSUE THREAD

**Issue by yinanfang** [Moya/Moya#384]
I'm struggling to get started because it seems like Result framework updated their syntax and the ReadMe example no longer works. So I was using the following code from the ReadMe but I got this error: "contextual type for closure argument list expects 1 argument but 4 were specified"

```
provider.request(.Test) { (data, statusCode, response, error) in
      if let data = data {
        // do something with the data
      }
    }
```

**Comment by ashfurrow** [comment]
Yikes! Yeah, our readme/documentation is out of date, sorry about that :disappointed: 

You can see a working example in the [Demo app](https://github.com/Moya/Moya/blob/a78b32e78c3db401c1038e3e819fca6105d9e830/Demo/Demo/ViewController.swift#L15). It's basically a Result type that's either success or failure. Let me know if I can clarify anything.

**Comment by yinanfang** [comment]
No problem. Thanks for point out! 
Do we have to use try and catch? I was usually told not to use try and catch in ObjC but now I see many try and catch statements in the swift, for example in this demo code. Is it a different code style or just a lazy way for making a simple demo app?

**Comment by ashfurrow** [comment]
try/catch is much much more common in Swift – it replaces Objectice-C's NSError model, actually. In this case, it's due to JSON parsing.

**Comment by yinanfang** [comment]
That makes sense! Thank you for the explanation. Moya is a great framework. Thank you for maintaining this!

**Comment by colinta** [comment]
Attempting fix w/ PR #388.

**Comment by ashfurrow** [comment]
Thanks @colinta !
